### ISSUE TITLE
ID3 from streaming (Icecast/SHOUTcast)

### ISSUE THREAD

**Issue by MisterGot** [aadsm/JavaScript-ID3-Reader#58]
Hi António,
Do you know if it's possible to decode the ID3 from a stream?
We plan to use the Web Audio API to decode the metadata but I don't see anything on the web talking about that.
Then  I found your solution but I guess it's only for FileAPI?
Thanks in advance,
Greg

**Comment by aadsm** [comment]
I've never tried but it should be possible. FileAPI is just one of the available readers, XHR and Node is also supported.
You'd need to implement your own reader though, but it should be possible since files are read asynchronously and in a random way meaning that we only load the bytes that are really needed (as opposed to reading the entire file).

I've been working on the next version of this library (here: https://github.com/aadsm/jsmediatags) and it should be much simpler to implement a new file reader there: https://github.com/aadsm/jsmediatags#new-file-readers.

**Comment by ManarKamel** [comment]
@MisterGot No, that's not possible, but there's a better way. Check this jQuery plugin: http://codecanyon.net/item/kast-extraordinary-shoutcast-html5-radio-player-for-jquery-material-design/14396614
